datasetinsights/io/tracker/mlflow.py

```diff
@@ -0,0 +1,101 @@
+import os
+import threading
+import time
+
+import mlflow
+from google.auth.transport.requests import Request
+from google.oauth2 import id_token
+
+
+class MLFlowTracker:
+    """ MlFlow tracker class, responsible for setting host, client_id and return
+        initialized mlflow. It also refreshes the access token through daemon
+        thread.
+    Examples:
+         # Set MLTracking server UI, default is local file
+        >>> mlflow.set_tracking_uri(TRACKING_URI)
+        # New run is launched under the current experiment
+        >>> mlflow.start_run()
+        # Log a parameter (key-value pair)
+        >>> mlflow.log_param("param_name", "param_value")
+        # Log a metric (key-value pair)
+        >>> mlflow.log_metric("metric_name", "metric_val")
+        # Log an artifact (output file)
+        >>> with open("output.txt", "w") as f:
+        >>>     f.write("Hello world!")
+        >>> mlflow.log_artifact("output.txt", "run1/output/")
+        # ends the run launched under the current experiment
+        >>> mlflow.end_run()
+    Attributes:
+        REFRESH_INTERVAL: default refresh token interval
+        __mlflow: holds initialized mlflow
+    """
+
+    REFRESH_INTERVAL = (
+        3000  # every 50 minutes refresh token. Token expires in 1 hour
+    )
+    __mlflow = None
+
+    def __init__(self, host_id, client_id, exp_name):
+        """constructor.
+        Args:
+            host_id: MlTracker server host
+            client_id: MLFlow tracking server client id
+            exp_name: name of the experiment
+        """
+        if client_id:
```

Comment:
shouldn't this fail immediately if there is no client id?

Reply:
- If this run is executed by unity employee then client id is required, in this case it will throw exception and NullTracker will be initiated.
- If this run is executed by non unity employ, it will try to connect to given tracking host if connected then will return mlflow, if any exception occur then NullTracker.

- Do you want MLflow Tracking feature only for unity employee?